tests/rptest/tests/create_partitions_test.py

```diff
@@ -6,24 +6,60 @@
 # As of the Change Date specified in that file, in accordance with
 # the Business Source License, use of this software will be governed
 # by the Apache License, Version 2.0
+import random
 
 from ducktape.mark.resource import cluster
+from ducktape.mark import ignore
+from ducktape.utils.util import wait_until
 from rptest.tests.redpanda_test import RedpandaTest
 
 from rptest.clients.types import TopicSpec
 from rptest.clients.kafka_cli_tools import KafkaCliTools
 
 
 class CreatePartitionsTest(RedpandaTest):
-    topics = (TopicSpec(partition_count=2, replication_factor=3), )
+    def _partition_count(self, topic):
+        client = KafkaCliTools(self.redpanda)
+        meta = client.describe_topic(topic)
+        return meta.partition_count
+
+    def _add_topic_partitions(self, topic, count):
+        client = KafkaCliTools(self.redpanda)
+        client.add_topic_partitions(topic, count)
+
+    def _create_add_verify(self, topic, new_parts):
+        self.logger.info(
+            f"Testing topic {topic.name} with partitions {topic.partition_count} replicas {topic.replication_factor} additional partitions {new_parts}"
+        )
+
+        self.redpanda.create_topic(topic)
+
+        wait_until(
+            lambda: self._partition_count(topic.name) == topic.partition_count,
+            timeout_sec=20,
+            backoff_sec=2,
+            err_msg=
+            f"Initial topic doesn't have expected {topic.partition_count} partitions, found {self._partition_count(topic.name)}"
+        )
+
+        self._add_topic_partitions(topic.name, new_parts)
+
+        expected_parts = topic.partition_count + new_parts
+        wait_until(
+            lambda: self._partition_count(topic.name) == expected_parts,
+            timeout_sec=20,
+            backoff_sec=2,
+            err_msg=
+            f"Initial topic doesn't have expected {expected_parts} partitions, found {self._partition_count(topic.name)}"
+        )
 
     @cluster(num_nodes=3)
     def test_create_partitions(self):
-        topic = self.topics[0].name
-        cli = KafkaCliTools(self.redpanda)
-        # add 5 partitions to the topic
-        cli.add_topic_partitions(topic, 5)
-
-        res = cli.describe_topic(topic)
-        # initially topic had 2 partitions, we added 5
-        assert res.partition_count == 7
+        big = self.scale.ci or self.scale.release
+        iterations = 12 if big else 2
```

Comment:
For example, why was 12 and 2 chosen as the number of iterations?

Reply:
I don't really have any guidance. It sufficient that 3 nodes probably got some action. It didn't take an exorbitant amount of time